types/datasource.d.ts

```diff
@@ -298,12 +298,94 @@ export declare class DataSource extends EventEmitter {
   ping(callback: Callback): void;
 
   // Only promise variant, callback is intentionally not supported.
+
+  /**
+   * Execute a SQL command.
+   *
+   * **WARNING:** In general, it is always better to perform database actions
+   * through repository methods. Directly executing SQL may lead to unexpected
+   * results, corrupted data, security vulnerabilities and other issues.
+   *
+   * @example
+   *
+   * ```ts
+   * // MySQL
+   * const result = await db.execute(
+   *   'SELECT * FROM Products WHERE size > ?',
+   *   [42]
+   * );
+   *
+   * // PostgreSQL
+   * const result = await db.execute(
+   *   'SELECT * FROM Products WHERE size > $1',
+   *   [42]
+   * );
+   * ```
+   *
+   * @param command A parameterized SQL command or query.
+   * Check your database documentation for information on which characters to
+   * use as parameter placeholders.
+   * @param parameters List of parameter values to use.
+   * @param options Additional options, for example `transaction`.
+   * @returns A promise which resolves to the command output as returned by the
+   * database driver. The output type (data structure) is database specific and
+   * often depends on the command executed.
+   */
   execute(
     command: string | object,
-    args?: any[] | object,
+    parameters?: any[] | object,
     options?: Options,
   ): Promise<any>;
 
+  /**
+   * Execute a MongoDB command.
+   *
+   * **WARNING:** In general, it is always better to perform database actions
+   * through repository methods. Directly executing MongoDB commands may lead
+   * to unexpected results and other issues.
+   *
+   * @example
+   *
+   * ```ts
+   * const result = await db.execute('MyCollection', 'aggregate', [
+   *   {$lookup: {
+   *     // ...
+   *   }},
+   *   {$unwind: '$data'},
+   *   {$out: 'tempData'}
+   * ]);
+   * ```
+   *
+   * @param collectionName The name of the collection to execute the command on.
+   * @param command The command name. See
+   * [Collection API docs](http://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html)
+   * for the list of commands supported by the MongoDB client.
+   * @param parameters Command parameters (arguments), as described in MongoDB API
+   * docs for individual collection methods.
+   * @returns A promise which resolves to the command output as returned by the
+   * database driver.
+   */
+  execute(
+    collectionName: string,
+    command: string,
+    ...parameters: any[],
```

Comment:
nitpick: the `execute` function in mongodb seems doesn't have a 3rd parameter, see 
https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L467

Reply:
I appreciate your attention to details! :+1:

MongoDB connector's `execute` function does not describe additional parameters in the function signature, but uses `arguments.slice` to access them:


https://github.com/strongloop/loopback-connector-mongodb/blob/c58e9fbe2b4161230751baf0385cb4c4f6958519/lib/mongodb.js#L469-L470

```js
  // Get the parameters for the given command
  const args = [].slice.call(arguments, 2);
```

This code was written before [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) were introduced to the language, therefore it was not possible to encode `...parameters` in the function signature.